### PR TITLE
chore: rollback rule

### PR DESCRIPTION
**Pull request by adityachoudhari26**
## Summary by CodeRabbit

* **New Features**
  * Added configurable automatic rollback: system can create and dispatch a rollback job to the last successful release when configured job statuses occur or when verification failures are enabled.
  * Policy rules can now include rollback settings to control which statuses or verification outcomes trigger rollbacks.
  * Rollback handling integrated into the release orchestration and status-change processing.

* **Tests**
  * Added comprehensive end-to-end tests for job- and verification-based rollback scenarios, policy matching, and edge cases.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: apps/workspace-engine/oapi/spec/schemas/policy.jsonnet [hunk 1]
```diff
@@ -235,6 +236,22 @@ local openapi = import '../lib/openapi.libsonnet';
     },
   },
 
+  RollbackRule: {
+    type: 'object',
+    properties: {
+      rollBackJobStatuses: {
```
Comment (jsbroks) [inline apps/workspace-engine/oapi/spec/schemas/policy.jsonnet]:
```suggestion
      onJobStatus: {
```
